persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/util/ParquetWriter.java

```diff
@@ -35,26 +35,76 @@
 import org.apache.spark.sql.*;
 import joptsimple.*;
 
+import java.util.Iterator;
+
+import static org.apache.spark.sql.functions.*;
+
 
 /**
  * Command Line tool to Write Parquet Files.
  */
 public class ParquetWriter {
 
-    private static void write(String inputFile, String outputFile) {
+    private static void write(String inputFile, String outputFile, String typeOfData) {
         SparkSession spark = SparkSession.builder()
             .appName("cBioPortal")
             .master("local[*]")
             .getOrCreate();
         
-        Dataset<Row> df = spark.read()
-            .format("csv")
-            .option("delimiter", "\t")
-            .option("header", "true")
-            .option("comment","#")
-            .load(inputFile);
-        df.write()
-            .mode("append").parquet(outputFile);
+        Dataset<Row> df = null;
+        Iterator it = null;
+        switch (typeOfData) {
+            case "case": case "panel":
+                df = spark.read()
+                    .format("csv")
+                    .option("delimiter", ":")
+                    .load(inputFile);
+
+                it = df.toLocalIterator();
+                while (it.hasNext()) {
+                    Row row = (Row) it.next();
+                    String colName = row.getString(0);
+                    String listCol = "case".equalsIgnoreCase(typeOfData) ? "case_list_ids" : "gene_list";
```

Comment:
Does this value ever change?

Reply:
i see both [case_list_ids](https://docs.cbioportal.org/5.1-data-loading/data-loading/file-formats#case-lists) and [gene_list](https://docs.cbioportal.org/5.1-data-loading/data-loading/import-gene-panels#gene-panel-file-format) is in the documentation.

Reply:
I mean once this routine is invoked, will listCol be assigned the same value each time through the while-loop?

Reply:
Thanks for clarifying. moved it out of the while-loop.